test/e2e/kubevirt.go

```diff
@@ -1420,4 +1429,145 @@ passwd:
 			}),
 		)
 	})
+	Context("with kubevirt VM using layer2 UDPN", func() {
+		var (
+			cidrIPv4                = "10.128.0.0/24"
+			cidrIPv6                = "2010:100:200::/60"
+			primaryUDNNetworkStatus nadapi.NetworkStatus
+			virtLauncherCommand     = func(command string) (string, error) {
+				stdout, stderr, err := ExecShellInPodWithFullOutput(fr, namespace, "virt-launcher-vm1", command)
+				if err != nil {
+					return "", fmt.Errorf("%s: %s: %w", stdout, stderr, err)
+				}
+				return stdout, nil
+			}
+			primaryUDNValueFor = func(field string) ([]string, error) {
+				output, err := virtLauncherCommand(fmt.Sprintf(`nmcli -e no -g %s c show ovn-udn1`, field))
+				if err != nil {
+					return nil, err
+				}
+				return strings.Split(output, " | "), nil
+			}
+		)
+		BeforeEach(func() {
+			netConfig := newNetworkAttachmentConfig(
+				networkAttachmentConfigParams{
+					namespace: namespace,
+					name:      "net1",
+					topology:  "layer2",
+					cidr:      correctCIDRFamily(cidrIPv4, cidrIPv6),
+					role:      "primary",
+				})
+			By("Creating NetworkAttachmentDefinition")
+			nad := generateNAD(netConfig)
+			Expect(crClient.Create(context.Background(), nad)).To(Succeed())
+
+			kubevirtPod := &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "virt-launcher-vm1",
+					Namespace: namespace,
+					Labels: map[string]string{
+						kubevirtv1.VirtualMachineNameLabel: "vm1",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:  "compute",
+						Image: "quay.io/nmstate/c10s-nmstate-dev:latest",
+						SecurityContext: &corev1.SecurityContext{
+							Privileged: ptr.To(true),
+							Capabilities: &corev1.Capabilities{
+								Add: []corev1.Capability{"NET_ADMIN"},
+							},
+						},
+					}},
+				},
+			}
+			By("Create virt-launcher pod")
+			Expect(crClient.Create(context.Background(), kubevirtPod)).To(Succeed())
+
+			By("Wait for virt-launcher pod to be ready and primary UDN network status to pop up")
+			waitForPodsCondition([]*corev1.Pod{kubevirtPod}, func(g gomega.Gomega, pod *corev1.Pod) {
+				ok, err := testutils.PodRunningReady(pod)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(ok).To(BeTrue())
+
+				primaryUDNNetworkStatuses, err := podNetworkStatus(pod, func(networkStatus nadapi.NetworkStatus) bool {
+					return networkStatus.Name == "ovn-kubernetes" && networkStatus.Default && networkStatus.Interface == "ovn-udn1"
+				})
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(primaryUDNNetworkStatuses).To(HaveLen(1))
```

Comment:
for this check to make sense, imho, you should not filter above by interface name, just by the `Default` flag.

Reply:
If something breaks in multus and we go back having eth0 as default or multiple defaults we may have a false possitive here, that's why the extra check is added here, at the end this is about primary UDN.

Reply:
AFAIU, for your argument to make sense, two things need to break in multus / status library:
- multiple default interfaces reported as default
- multiple interfaces reported with the same interface name

I'm simply saying the second criteria is safe to assume, since it would also mean the CNIs are returning the same interface twice in the CNI result.

Reply:
Ok, will filter just by default flag.